aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -307,8 +334,12 @@ module aptos_framework::fungible_asset {
 
     #[view]
     /// Get the balance of a given store.
-    public fun balance<T: key>(store: Object<T>): u64 acquires FungibleStore {
-        if (store_exists(object::object_address(&store))) {
+    public fun balance<T: key>(store: Object<T>): u64 acquires FungibleStore, ConcurrentFungibleBalance {
+        let store_addr = object::object_address(&store);
+        if (concurrent_fungible_balance_exists(store_addr)) {
+            let balance_resource = borrow_global<ConcurrentFungibleBalance>(store_addr);
+            aggregator_v2::read(&balance_resource.balance)
+        } else if (store_exists(store_addr)) {
```

Comment:
How about using `concurrent_fungible_balance_exists_inline` and `store_exists_inline` as suggested above to eliminate gas bloat?

Reply:
you are so paranoid, hah

Reply:
@alnoki has a point, we have many single-block functions, which are extra costs and extra performance losses. Why not inline if we could